src/ophys_etl/modules/segmentation/modules/schemas.py

```diff
@@ -465,6 +475,29 @@ class RoiMergerSchema(argschema.ArgSchema):
         description=("fraction of timesteps to keep when doing time "
                      "correlations"))
 
+    @post_load
+    def check_for_rois(self, data, **kwargs):
+        """rois can be passed in as a path to json in 'roi_input'
+        or can be read from within the in-progress 'qc_output'
+        """
+        if data["roi_input"] is None:
+            if data["qc_output"] is None:
+                raise ValidationError(
+                        "either 'roi_input' (json) must be specified "
+                        "or 'qc_output' (hdf5) with group 'detect' with "
+                        "dataset 'rois' must be specified.")
+                with h5py.File(data["qc_output"], "r") as f:
```

Comment:
Looks like there is an indent error here. I think this block only executes if `data['qc_output']` is `None`

Reply:
fixed, cleaned up, added test.